src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,160 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.RegistrationServiceImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    void register_nullValue_notOk() {
+        User user = new User();
+        user.setLogin("12345");
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_loginLengthFiveOrLessCharacters_notOk() {
+        User user = new User();
+        user.setLogin("12345");
+        user.setPassword("userpassword");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_loginLengthMoreFiveCharacters_Ok() {
+        User user = new User();
+        user.setLogin("123456");
+        user.setPassword("userpassword");
+        int actual = user.getLogin().length();
+        int expectLoginLength = 6;
+        assertTrue(expectLoginLength <= actual);
+    }
+
+    @Test
+    void storageDao_addUserToStorage_ok() {
+        User user = new User();
+        user.setId(1L);
+        user.setPassword("1234562");
+        user.setLogin("Cat");
+        user.setAge(19);
+        storageDao.add(user);
+        User actual = storageDao.get("Cat");
+        assertEquals(user, actual);
```

Comment:
Tests in this class should test methods of registration service, you basically test storageDao